matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/state/StateService.kt

```diff
@@ -68,8 +68,11 @@ interface StateService {
 
     /**
      * Send a state event to the room
+     * @param eventType The type of event to send.
+     * @param stateKey The state_key for the state to send. Can be an empty string.
+     * @param body The content object of the event; the fields in this object will vary depending on the type of event
      */
-    suspend fun sendStateEvent(eventType: String, stateKey: String?, body: JsonDict)
+    suspend fun sendStateEvent(eventType: String, stateKey: String, body: JsonDict)
```

Comment:
are there other endpoints which use empty strings to represent the absence of a value? 

thinking out loud, from a public api perspective I would prefer to pass a null than an empty string as it gives us some extra type safety  (and then let the internal implementation convert to an empty string)

Reply:
We want the SDK API to follow as much as possible the Matrix specification/philosophy. In the spec, the `stateKey` cannot be null.
Ref: https://matrix.org/docs/spec/client_server/r0.6.1#put-matrix-client-r0-rooms-roomid-state-eventtype-statekey

Reply:
> extra type safety

I do not understand why?

Reply:
a string can contain content or be empty, whereas a null can only be null

the spec is a little bit misleading here when it mentions empty strings because the value of "" is an implementation detail of building a url path, we never send `""` to the api.

it actually means to skip applying the `stateKey` segment, which means the `stateKey required` is actually a lie...

```
has a state key
/_matrix/client/r0/rooms/{roomId}/state/{eventType}/{stateKey}/

doesn't have a state key
/_matrix/client/r0/rooms/{roomId}/state/{eventType}/
```

assuming I've understood correctly! not a blocker for the PR, something I found interesting.

Reply:
In API paths we do not send "" (or we alsways do it, since its nothing!), but I am wondering if 
`/_matrix/client/r0/rooms/{roomId}/state/{eventType}/` is equivalent to
`/_matrix/client/r0/rooms/{roomId}/state/{eventType}` (note the missing trailing `/`).
Anyway I think the spec make a diff mainly for the Json of the Event content itself:
`"state_key": ""` is not equivalent to `"state_key": null` (or missing `"state_key"`).
that's maybe why it is a bit confusing all along.